### ISSUE TITLE
Frame parsing section not complete

### ISSUE THREAD

**Issue by shaqtsui** [Abc-Arbitrage/fixpp#9]
Frame parsing section not complete, can not have a glimps on the parsing usage.

**Comment by oktal** [comment]
Hello,

You are right, the documentation is not really exhaustive on that part.

In the meantime,  here is  how you should use the API to parse a FIX frame.

1. First, define your visit rules:

```cpp
struct VisitRules : public Fixpp::VisitRules
{
    using Overrides = OverrideSet<>;

    // Visitor will only handle FIX 4.2 messages
    using Dictionary = Fixpp::v42::Spec::Dictionary;

    // Validate the checksum
    static constexpr bool ValidateChecksum = true;

    // Validate the length of the frame
    static constexpr bool ValidateLength = true;

    // Do not parse in strict mode (e.g do not stop on an unknown tag)
    static constexpr bool StrictMode = false;

    // Do not skip unknown tags
    static constexpr bool SkipUnknownTags = false;
};
```

2. Define your visitor and override the `operator()`  function call to handle the messages you want. For example, this will handle the `Logon`  message

```cpp
struct Visitor : public Fixpp::StaticVisitor<void>
{
    void operator()(const Fixpp::v42::Header::Ref& header, const Fixpp::v42::Message::Logon::Ref& logon)
    {
        ASSERT_EQ(Fixpp::get<Fixpp::Tag::SenderCompID>(header), "ABC");
        ASSERT_EQ(Fixpp::get<Fixpp::Tag::HeartBtInt>(logon), 60);

        Fixpp::get<Fixpp::Tag::SendingTime>(header);
    }

    template<typename HeaderT, typename MessageT> void operator()(HeaderT, MessageT)
    {
    }
};
```

3. Call `Fixpp::visit`  with your rules and your visitor:

```cpp
const char *frame = "[here goes your raw frame]" 
const size_t size = std::strlen(frame);

Visitor visitor;

Fixpp::visit(frame, size, visitor, VisitRules()).otherwise([&](const Fixpp::ErrorKind& e) {
    // Handle error
    auto errStr = e.asString();
});
```

Let me know how it goes

**Comment by shaqtsui** [comment]
Thanks a lot for detailed explaination.
